make/hotspot/lib/CompileJvm.gmk

```diff
@@ -61,11 +61,6 @@ else
   OPENJDK_TARGET_CPU_VM_VERSION := $(OPENJDK_TARGET_CPU)
 endif
 
-ifeq ($(VERSION_BUILD), )
-  # Hotspot cannot handle an empty build number
-  VERSION_BUILD := 0
-endif
```

Comment:
I think the proposed "solution" is *much* worse than this.

Reply:
Reverted to use the original, less intrusive solution from [8274980](https://github.com/openjdk/jdk/pull/11081/commits/83ed3deb29d7344bbc95a3831f2388d077bc59e9) that initially could not work with the older Visual C++ compiler (With a minor improvement to handle #define 0)

Reply:
Sorry, but I don't think that's much better than the prior version, and still doesn't seem better than the current code.  What problem is this change supposed to be solving?  I didn't find any open bugs that seemed relevant (could be I just didn't recognize such).  Whatever it is seems likely to be unrelated to any of the other changes in this PR; it would have been / would be better to deal with it separately.

Reply:
It's related to the JEP-223 version build numbers, which the original commit couldn't get to work due to a preprocessor trick not working with the now unsupported 2017. I just felt it could be included here out of convenience, but I guess I can split it off into another change if required